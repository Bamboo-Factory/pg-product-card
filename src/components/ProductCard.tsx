import React, { createContext, CSSProperties } from 'react';
import { useProduct } from '../hooks/useProduct';
import {
  InitialValuesProps,
  OnChangeArgs,
  Product,
  ProductContextProps,
} from '../interfaces/product.interfaces';

import styles from '../styles/styles.module.css';
import { ProductCardHandlers } from '../interfaces/product.interfaces';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export interface Props {
  product: Product;
  // children?: ReactElement | ReactElement[];
  children: (args: ProductCardHandlers) => JSX.Element;
  className?: string;
  style?: CSSProperties;
  onChange?: (args: OnChangeArgs) => void;
  value?: number;
  initialValues?: InitialValuesProps;
}

export const ProductCard = ({
  children,
  product,
  className,
  style,
  onChange,
  value,
  initialValues,
}: Props) => {
  const {
    counter,
    increaseBy,
    maxCount,
    isMaxCountReached,
    reset,
  } = useProduct({
    onChange,
    product,
    value,
    initialValues,
  });

  return (
    <Provider value={{ counter, increaseBy, product, maxCount }}>
      <div className={`${styles.productCard} ${className}`} style={style}>
        {children({
          count: counter,
          isMaxCountReached,
          maxCount: initialValues?.maxCount,
          product,
          increaseBy,
          reset,
        })}
      </div>
    </Provider>
  );
};
