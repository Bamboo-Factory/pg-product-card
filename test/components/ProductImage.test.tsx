import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';
describe('ProductImage', () => {
  test('show component with custom image', () => {
    const wrapper = renderer.create(<ProductImage img="http://product.jpg" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('show component with product image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
