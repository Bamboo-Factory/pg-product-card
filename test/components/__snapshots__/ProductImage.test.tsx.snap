// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage show component with custom image 1`] = `
<img
  alt="product"
  className="productImg undefined"
  src="http://product.jpg"
/>
`;

exports[`ProductImage show component with product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
